pkg/controller/service_test.go

```diff
@@ -0,0 +1,383 @@
+package controller_test
+
+import (
+	"context"
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/containous/maesh/pkg/controller"
+	"github.com/containous/maesh/pkg/k8s"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	listers "k8s.io/client-go/listers/core/v1"
+)
+
+type tcpPortMapperMock struct {
+	findFunc func(svc k8s.ServiceWithPort) (int32, bool)
+	getFunc  func(srcPort int32) *k8s.ServiceWithPort
+	addFunc  func(svc *k8s.ServiceWithPort) (int32, error)
+}
+
+func (t tcpPortMapperMock) Find(svc k8s.ServiceWithPort) (int32, bool) {
+	return t.findFunc(svc)
+}
+
+func (t tcpPortMapperMock) Get(srcPort int32) *k8s.ServiceWithPort {
+	return t.getFunc(srcPort)
+}
+
+func (t tcpPortMapperMock) Add(svc *k8s.ServiceWithPort) (int32, error) {
+	return t.addFunc(svc)
+}
+
+func Test_ServiceCreate(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    corev1.Service
```

Comment:
could you rename this to `provided` to keep consistent with the other unit tests?